api/control.proto

```diff
@@ -183,6 +184,7 @@ message ListNodesRequest {
 	}
 
 	Filters filters = 1;
+	bool available_resources = 2;
```

Comment:
I don't think you need to plumb this flag. Just include the resources with the standard request.

If that isn't going to work, we need to come up with a more generic way to handle selective inclusion of data, such as field paths. This approach gets nasty if you just keep adding booleans.

Reply:
@stevvooe the challenge is this operation need calculate the available resources by tasks status. It is not a simple filtering. I add the flag just to avoid the performance impact for simple listing/getting node status

Reply:
@denverdino Yes, there is expense. Right now, we don't have a selective inclusion model when fetching resources. Adding booleans for each addition will become cumbersome. The node endpoint is supposed to return the node's data, as is, so unless the bookkeeping is kept directly on the node, this isn't the right model.

In the past, we don't really have these endpoints do calculations of this sort. This really something that we have pushed to the client in the past. If that would work, this might be better as a separate endpoint.

Reply:
@stevvooe Such calculation in client side will introduce many cost for network, it is better in server-side. And It is a pretty common requirement, most of user don't know how to do that properly.

Reply:
I feel like a boolean to control inclusion/exclusion of an expensive calculated field is cleaner than introducing a new set of endpoints. We could possibly formalize this with some kind of `Features` message that contains feature flags for the request.

Reply:
@aaronlehmann The goal of our endpoints has always been to returned data structure as seen by the orchestrator. Having a calculated reporting value is a new model and we need to consider the impacts of it for performance and behavior.

I don't see the issue with adding a new method to get the nodes orchestration statistics.

@denverdino What is the actual expense? Do you have numbers showing the performance difference?

Reply:
the expense is related to the number of tasks.

Reply:
@denverdino Can you quantify it?

Reply:
Nope, but if we have hundreds of tasks on each node it will take pretty much time to fetch tasks, and filtering. If we do that in client side it will take moretime on serialization and transportation

Reply:
Ok, if client side is not an option, and I'm not convinced that its not, because you haven't benchmarked it, the options are the following:

1. Build out a model for selective inclusion of fields that will scale to other types.
2. Have an endpoint that focuses on providing aggregate views.

Let's not sacrifice the design constraints for expediency. The above are reasonable approaches to this problems. I'm in favor of #2, as I think it complements the the current API model.

Reply:
@stevvooe any suggestion for the endpoint? thx

Reply:
`GetNodeResourceUsage` or `GetNodeStatus`. I think the first step here is to look at what this endpoint might encompass. Basically, we are trying to see some of the aggregates that the orchestrator sees but are not obvious from the single node record. What are some other facts about orchestration state that we may want to expose?